Polyhedron/demo/Polyhedron/include/CGAL/IO/read_surf_trianglemesh.h

```diff
@@ -158,20 +159,23 @@ void read_surf(std::istream& input, std::vector<Mesh>& output, std::vector<Mater
       iss.clear();
       iss.str(line);
       iss >> index[0] >> index[1] >> index[2];
+      CGAL::cpp11::array<vertex_descriptor, 3> face;
       for(int id=0; id<3; ++id)
       {
-        if(vertices.find(index[id]) == vertices.end())
+        if(vertices[index[id]] == null_vertex)
         {
-          vertices.insert(std::make_pair(index[id], add_vertex(points[index[id]-1], mesh)));
+          vertices[index[id]] = add_vertex(points[index[id]-1], mesh);
         }
+        face[id] = vertices[index[id]];
       }
```

Comment:
... and here, if `vertices[index[id]] == null_vertex`, that means that vertex with ID `index[id]` is a new one.